src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -95,6 +96,11 @@ private static bool BeforeExecuteCommand(bool speculative, TypeCharCommandArgs a
                 return false;
             }
 
+            if (!args.SubjectBuffer.GetFeatureOnOffOption(FeatureOnOffOptions.MoveToEndOfExpressionOnSemiColon))
+            {
+                return false;
+            }
```

Comment:
I don't know, but if this call is cheaper than `GetOpenDocumentInCurrentContextWithChanges`, probably move it before?

Reply:
yes, this is cheaper and shoudl be done above if possible.

Reply:
technically, it's the same cost.  but it likely is clearer to move earlier.